### PR TITLE
Added Comment serializer and views methods

### PR DESCRIPTION
**Pull request by BarBoley**
Added Comment's views and methods:get,put,delete,post

### CODE REVIEW

File: iconic/UserProfile/views.py [hunk 1]
```diff
@@ -173,17 +174,72 @@ def put(self, request):
...
+    def post(self, request):
+        data = dict(request.data)
+        serializer = CommentSerializer(data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data,
+                            status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: This code want from me to put user_id in request, but the task was request, that takes request.user.id like user_id

File: iconic/UserProfile/views.py [hunk 2]
```diff
@@ -173,17 +174,72 @@ def put(self, request):
...
+    def put(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        data = dict(request.data)
+        serializer = CommentSerializer(comment, data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Same problem as post method

Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Same problem as post method

File: iconic/UserProfile/views.py [hunk 3]
```diff
@@ -173,17 +174,72 @@ def put(self, request):
...
+                            status=status.HTTP_201_CREATED)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # Method for changing comment
+    def put(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        data = dict(request.data)
+        serializer = CommentSerializer(comment, data=data)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: A comment can only be changed by the user who created it, but in this code, anyone who is logged in can do it

Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: A comment can only be changed by the user who created it, but in this code anyone can do it

File: iconic/UserProfile/views.py [hunk 4]
```diff
@@ -173,17 +174,72 @@ def put(self, request):
...
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
+
+    # Method for deleting comment by id
+    def delete(self, request):
+        Comment.objects.get(id=request.data['id']).delete()
+        return Response(status=status.HTTP_200_OK)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Only creator of comment should be able to delete comment

Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: Only creator of comment should be able to delete comment

File: iconic/UserProfile/views.py [hunk 5]
```diff
@@ -173,17 +174,72 @@ def put(self, request):
...
+
+
+# View for getting all comments
+class CommentsGet(APIView):
+
+    # Method for getting all comments
+    def get(self, request):
+        comments = Comment.objects.filter(post_id=request.data['post_id'])
+        serializer = CommentSerializer(comments, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: This request shoul be unauthorized

Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: This request shoul be unauthorized

**Review by Hrustyaschiy — CHANGES_REQUESTED** [review]

File: iconic/UserProfile/views.py [hunk 6]
```diff
@@ -173,17 +174,76 @@ def put(self, request):
...
+        comment = Comment.objects.get(id=request.data['id'],
+                                      user_id=request.user.id)
+        data = dict(request.data)
+        serializer = CommentSerializer(comment, data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: this code still want from me user_id (just put it in data like you did it in post request), anyway there is one another problem: this code get post_id but just put it in database, and the task was request that change comment that identified by id and post_id, but you do not use post_id to identify comment

File: iconic/UserProfile/views.py [hunk 7]
```diff
@@ -173,17 +174,76 @@ def put(self, request):
...
+    # Method for getting comment by id
+    def get(self, request):
+        comment = Comment.objects.get(id=request.data['id'])
+        serializer = CommentSerializer(comment)
+        if serializer.is_valid():
+            return Response(serializer.data,
+                            status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST)
```
Comment (Hrustyaschiy) [inline iconic/UserProfile/views.py]: I now thought, and decided that this request must be unauthorized
Make:
comments/ url for this one
comments/all/ for getting all comments of the post
comments/edit/ for post,put and delete
Also you do not need to check if the data is valid (line 201) just return Response(serializer.data, status=status.HTTP_200_OK)

**Review by Hrustyaschiy — CHANGES_REQUESTED** [review]

**Review by Hrustyaschiy — APPROVED** [review]

**Review by Hordd-Scratch — APPROVED** [review]

**Review by Ytachkin — APPROVED** [review]

**Pull request merged**